utils/misc.mli

```diff
@@ -38,9 +38,30 @@ val split_last: 'a list -> 'a list * 'a
 val samelist: ('a -> 'a -> bool) -> 'a list -> 'a list -> bool
         (* Like [List.for_all2] but returns [false] if the two
            lists have different length. *)
-
+val sameoption: ('a -> 'a -> bool) -> 'a option -> 'a option -> bool
+val map2_head: ('a -> 'b -> 'c) -> 'a list -> 'b list -> ('c list * 'b list)
```

Comment:
`head` denotes the first element of a list. This would be `map2_prefix`, but I am bit surprised that the definition is so asymmetric. A more general return type would be ``'c list * [ `LeftSuffix of 'a list | `RightSuffix of 'b list ]``. I think that either having the more general version or *not* having a function at all (using `length`, `split_at` and `map2` explicitly) would make the call-site more readable. Is this operation time-critical, or can we afford the extra traversals of the `length` + `split_at` solution?